src/main/resources/xslt/schemaorg2solr.xsl

```diff
@@ -97,31 +98,31 @@
           <!-- Extract the creater affiliation. Two fields are required here as creator_affiliation can change over time.
                Therefore, we are also extracting the creator_affiliation_generic which contains the same value for e.g.
                DR P1 from 1960 'program 1' and 2000's 'P1'. Here the value would be drp1. -->
-          <!-- map:find() can be used, because we know that only one key in the complete JSON file is named
-               broadcastDisplayName -->
           <xsl:if test="f:exists(map:get($schemaorg-xml, 'publication'))">
-            <xsl:if test="f:exists(map:get(map:get($schemaorg-xml, 'publication'),'publishedOn'))">
+            <xsl:if test="f:exists(my:getNestedMapValue($schemaorg-xml, 'publication','publishedOn'))">
 
-              <xsl:if test="f:exists(map:get(map:get(map:get($schemaorg-xml, 'publication'),'publishedOn'), 'broadcastDisplayName' ))">
-                <xsl:if test="f:exists(map:find($schemaorg-xml,'broadcastDisplayName'))">
+              <xsl:if test="not(f:empty(my:getTripleNestedMapValue($schemaorg-xml, 'publication', 'publishedOn',  'broadcastDisplayName')))">
                   <f:string key="creator_affiliation">
-                    <xsl:value-of select="map:find($schemaorg-xml,'broadcastDisplayName')"/>
+                    <xsl:value-of select="my:getTripleNestedMapValue($schemaorg-xml, 'publication', 'publishedOn',  'broadcastDisplayName')"/>
                   </f:string>
-                </xsl:if>
               </xsl:if>
 
-              <xsl:if test="f:exists(map:get(map:get(map:get($schemaorg-xml, 'publication'),'publishedOn'), 'alternateName'))">
+              <xsl:if test="not(empty(my:getTripleNestedMapValue($schemaorg-xml, 'publication', 'publishedOn', 'alternateName')))">
                 <f:string key="creator_affiliation_generic">
-                  <xsl:value-of select="map:get(map:get(map:get($schemaorg-xml, 'publication'),'publishedOn'), 'alternateName')"/>
+                  <xsl:value-of select="my:getTripleNestedMapValue($schemaorg-xml, 'publication', 'publishedOn', 'alternateName')"/>
+                </f:string>
+              </xsl:if>
+
+              <xsl:if test="not(f:empty(my:getFourLevelNestedMapValue($schemaorg-xml, 'publication',
+                                                        'publishedOn', 'broadcaster', 'legalName')))">
+                <f:string key="broadcaster">
+                  <xsl:value-of select="my:getFourLevelNestedMapValue($schemaorg-xml, 'publication',
+                                                          'publishedOn', 'broadcaster', 'legalName')"/>
                 </f:string>
```

Comment:
Would it be possible to extend the helper methods even further so that they also do the "if exists then set the key"-step as that if a very recurring pattern?

Reply:
It would only be possible to return the value as they already do. This is because the key attribute on strings, arrays etc. doesn't accept variables. We have to define the key manually. I could create quite specific if statements to handle this, but I don't think that would make anything prettier. It would just move the mess to util.xsl